File: EduMS.Backend/src/EduMS.Application/M6_StatisticsReports/DTOs/StatisticsArchives/StatisticsArchiveDto.cs
using System;

namespace EduMS.Application.M6_StatisticsReports.DTOs.StatisticsArchives;

public class StatisticsArchiveDto
{
    public long Id { get; set; }
    public long SubmittedStatisticsId { get; set; }
    public long SchoolId { get; set; }
    public string ArchivedYear { get; set; } = string.Empty;
    public int PeriodType { get; set; }
    public DateTime ArchivedAt { get; set; } = DateTime.UtcNow;
    public long ArchivedByUserId { get; set; }
    public string? FinalDataSnapshotJson { get; set; }
    public string? StudentSnapshotJson { get; set; }
    public string? StaffSnapshotJson { get; set; }
    public int RetentionPeriodYears { get; set; } = 10;
    public DateTime? RetentionEndDate { get; set; }
    public bool IsReadOnly { get; set; } = true;
    public string? Notes { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public long CreatedByUserId { get; set; }
    public DateTimeOffset? ModifiedAt { get; set; }
    public long? ModifiedByUserId { get; set; }
    public bool IsDeleted { get; set; }
    public DateTimeOffset? DeletedAt { get; set; }
    public long? DeletedByUserId { get; set; }
    public Guid VersionToken { get; set; }
    public DateTimeOffset? LastSyncedAt { get; set; }
    public string SyncStatus { get; set; } = string.Empty;
}

File: src/app/modules/m6-StatisticsReports/data-access/models/statistics-archive.dto.ts
import { BaseAuditFields, PeriodType } from './base.types';

export interface StatisticsArchive extends BaseAuditFields {
  submittedStatisticsId: number;
  schoolId: number;
  archivedYear: string;
  periodType: PeriodType;
  archivedAt: string;
  archivedByUserId: number;
  finalDataSnapshotJson: string | null;
  studentSnapshotJson: string | null;
  staffSnapshotJson: string | null;
  retentionPeriodYears: number;
  retentionEndDate: string | null;
  isReadOnly: boolean;
  notes: string | null;
}

export type CreateStatisticsArchive = Omit<StatisticsArchive, 'id' | 'createdAt' | 'modifiedAt'>;

export type UpdateStatisticsArchive = Pick<StatisticsArchive, 'id'> & Partial<Omit<StatisticsArchive, 'id' | 'createdAt' | 'modifiedAt'>>;
